grow-with-freya/store/app-store.ts

```diff
@@ -174,10 +176,14 @@ export const useAppStore = create<AppState>()(
         notificationsEnabled: state.notificationsEnabled,
         hasRequestedNotificationPermission: state.hasRequestedNotificationPermission,
         textSizeScale: state.textSizeScale,
-
         backgroundAnimationState: state.backgroundAnimationState,
       }),
-
+      onRehydrateStorage: () => (state) => {
```

Comment:
There's a potential race condition here. If the app checks `hasHydrated` before the `onRehydrateStorage` callback fires, it will be false even though hydration might be complete. The initial state sets hasHydrated: false, but there's no guarantee when this callback executes.